File: src/com/ptby/dynamicreturntypeplugin/signatureconversion/StringList.java
package com.ptby.dynamicreturntypeplugin.signatureconversion;

import java.util.ArrayList;
import java.util.Iterator;

public class StringList extends ArrayList<String> {

    public StringList( String... strings ) {
        for ( String string : strings ) {
            add( string );
        }
    }


    public String toString() {
        String output = "";

        for ( String s : this ) {
            output += "signature : " + s + "\n";
        }

        return "[" + "\n" + output + "]";
    }

}

File: src/main/kotlin/com/ptby/dynamicreturntypeplugin/signatureconversion/StringList.kt
package com.ptby.dynamicreturntypeplugin.signatureconversion

import java.util.*

class StringList(vararg strings: String) : ArrayList<String>() {

    init {
        for (string in strings) {
            add(string)
        }
    }


    override fun toString(): String {
        var output = ""

        for (s in this) {
            output += "signature : " + s + "\n"
        }

        return "[\n$output]"
    }

}
